src/hooks/use-local-storage-state.ts

```diff
@@ -1,12 +1,69 @@
-import {Dispatch, SetStateAction, useEffect, useState} from 'react';
-import {emptyBoard} from '../services/game.service';
+import { useEffect, useState } from 'react';
+import { emptyBoard } from '../services/game.service';
 
-export const useLocalStorageState = <T>(key: string, initialState: T): [T, Dispatch<SetStateAction<T>>] => {
-  const [state, setState] = useState<T>(() => JSON.parse(localStorage.getItem(key) || JSON.stringify(emptyBoard)) || initialState);
+interface useLocalStorageReturnState<T> {
+    addToSquareHistory: (x: T) => void,
+    currentSquare: T;
+    moveBack: () => void,
+    moveForward: () => void,
+    reset: () => void,
+};
 
-  useEffect(() => {
-    localStorage.setItem(key, JSON.stringify(state));
-  }, [key, state]);
+export const useLocalStorageState = <T = string[]>(key: string, indexKey: string, initialState: T): useLocalStorageReturnState<T> => {
```

Comment:
I don't think we should tightly couple useLocalStorage state with state history, I would do useStateHistory as separate hook, then every hook would have it's own responsibility

Reply:
I would just use that useLocalStorageState in proper place and then it will be more generic as instead of `addToSquareHistory` function we would have something like `setState` function and it's a lot more generic then.

Reply:
Anyway you are settings state inside of `addToSquareHistory` function and you can do it somewhere else